cmd/buildctl/common/common.go

```diff
@@ -21,9 +23,40 @@ func ResolveClient(c *cli.Context) (*client.Client, error) {
 		}
 		serverName = uri.Hostname()
 	}
-	caCert := c.GlobalString("tlscacert")
-	cert := c.GlobalString("tlscert")
-	key := c.GlobalString("tlskey")
+
+	var caCert string
+	var cert string
+	var key string
+
+	tlsDir := c.GlobalString("tlsdir")
+
+	if tlsDir != "" {
+		// Look for ca.pem and, if it exists, set caCert to that
+		// Look for cert.pem and, if it exists, set key to that
+		// Look for key.pem and, if it exists, set tlsDir to that
+		for _, v := range [3]string{"/ca.pem", "/cert.pem", "/key.pem"} {
+			matches, _ := filepath.Glob(tlsDir + v)
```

Comment:
This looks like a weird construction to me.

```
if _, err := os.Stat(filepath.Join(tlsDir, v)); err != nil {
  switch v {
```
seems more readable to me. Also, I think we need to error if one of the files can't be found under `tlsdir`

Reply:
Gotcha, applying your changes and adding an error if one of the files are not found in the specified TLS directory